Sources/StreamVideoSwiftUI/Utils/PictureInPicture/StreamPictureInPictureVideoRenderer.swift

```diff
@@ -113,45 +113,50 @@ final class StreamPictureInPictureVideoRenderer: UIView, RTCVideoRenderer {
     // MARK: - Rendering lifecycle
 
     /// This method is being called from WebRTC and asks the container to set its size to the track's size.
-    func setSize(_ size: CGSize) {
-        trackSize = size
-    }
-
-    func renderFrame(_ frame: RTCVideoFrame?) {
-        guard let frame = frame else {
-            return
+    nonisolated func setSize(_ size: CGSize) {
+        Task { @MainActor in
+            self.trackSize = size
         }
+    }
 
-        // Update the trackSize and re-calculate rendering properties if the size
-        // has changed.
-        trackSize = .init(width: Int(frame.width), height: Int(frame.height))
+    nonisolated func renderFrame(_ frame: RTCVideoFrame?) {
+        nonisolated(unsafe) let unsafeFrame = frame
+        Task { @MainActor [weak self] in
```

Comment:
I'm not sure that running the decoding part on MainActor is wise. What was the part that was complaining? Can we reduce the MainActor calling and move to the background for the transformation?

Reply:
there were some things, but I made them nonisolated.